packages/vm/lib/evm/opFns.ts

```diff
@@ -676,16 +679,23 @@ export const handlers: { [k: string]: OpHandler } = {
     if (runState._common.gteHardfork('spuriousDragon')) {
       // We are at or after Spurious Dragon
       // Call new account gas: account is DEAD and we transfer nonzero value
-      if ((await runState.stateManager.accountIsEmpty(toAddressBuf)) && !value.isZero()) {
+      if ((await runState.eei.isAccountEmpty(toAddressBuf)) && !value.isZero()) {
         runState.eei.useGas(new BN(runState._common.param('gasPrices', 'callNewAccount')))
       }
-    } else if (!(await runState.stateManager.accountExists(toAddressBuf))) {
-      // We are before Spurious Dragon
+    } else if (!(await runState.eei.accountExists(toAddressBuf))) {
+      // We are before Spurious Dragon and the account does not exist.
       // Call new account gas: account does not exist (it is not in the state trie, not even as an "empty" account)
-      const accountDoesNotExist = !(await runState.stateManager.accountExists(toAddressBuf))
-      if (accountDoesNotExist) {
-        runState.eei.useGas(new BN(runState._common.param('gasPrices', 'callNewAccount')))
-      }
+      runState.eei.useGas(new BN(runState._common.param('gasPrices', 'callNewAccount')))
+    }
+
+    gasLimit = maxCallGas(
```

Comment:
Note that `maxCallGas` is called after we have deducted all the call gas (this was not done before, `callNewAccount` could be deducted after `maxCallGas`). This was a difference from geth, so this might actually fix some consensus issues if we would do a mainnet sync (although it is very strange that no state test has found this!). The issue is thus that the forwarded gas between geth/EthereumJS might be different.